src/mixins/formWatchers.js

```diff
@@ -21,10 +22,15 @@ export default {
     /**
      * Watch data changes
      * @param {object} data
+     * @param {array} changes
      */
-    watchDataChanges(data) {
+    watchDataChanges(data, changes) {
       if (this.watchers && this.watchers instanceof Array) {
-        this.watchers.forEach(watcher => this.checkWatcher(watcher, data));
+        this.watchers.forEach(watcher =>  {
+          if (changes.indexOf(watcher.watching) !== -1) {
```

Comment:
Use `changes.includes(watcher.watching)` instead of `indexOf` as it's more readable: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#:~:targetText=The%20includes()%20method%20determines,true%20or%20false%20as%20appropriate.